File: backend/src/main/java/es/udc/paproject/backend/model/services/CatalogService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Province;
import es.udc.paproject.backend.model.entities.SportEvent;
import es.udc.paproject.backend.model.entities.SportEventType;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;

import java.time.LocalDate;
import java.util.List;

public interface CatalogService {
    List<Province> findAllProvinces();
    List<SportEventType> findAllSportEventTypes();
    Block<SportEvent> findSportEvent(Long provinceId, Long sportEventTypeId,
                                     LocalDate dateStart, LocalDate dateEnd, int page, int number);
    SportEvent getSportEvent(Long sportEventId)
            throws InstanceNotFoundException;
}

File: frontend/src/backend/catalogService.js
import {config, appFetch} from './appFetch';

export const findAllProvinces = (onSuccess) =>
    appFetch('/catalog/provinces', config('GET'), onSuccess);

export const findAllSportEventTypes = (onSuccess) =>
    appFetch('/catalog/types', config('GET'), onSuccess);

export const getSportEvent = (id, onSuccess) =>
    appFetch(`/catalog/sportevents/${id}`, config('GET'), onSuccess);

export const findSportEvent = ({provinceId, sportEventTypeId, dateStart, dateEnd, page},
                               onSuccess) => {

    let path = `/catalog/sportevents?page=${page}`;

    path += provinceId ? `&provinceId=${provinceId}` : "";
    path += sportEventTypeId ? `&sportEventTypeId=${sportEventTypeId}` : "";
    path += dateStart ? `&dateStart=${dateStart}` : "";
    path += dateEnd ? `&dateEnd=${dateEnd}` : "";

    appFetch(path, config('GET'), onSuccess);

}